azurerm/resource_arm_iothub.go

```diff
@@ -621,6 +690,44 @@ func expandIoTHubRoutes(d *schema.ResourceData) *[]devices.RouteProperties {
 	return &routeProperties
 }
 
+func expandIoTHubFileUpload(d *schema.ResourceData) (map[string]*devices.StorageEndpointProperties, map[string]*devices.MessagingEndpointProperties, bool, error) {
+	fileUploadList := d.Get("file_upload").([]interface{})
```

Comment:
Note the return signature. The SDK spreads the file upload configuration across several structures. If desired, I can maje this three separate functions to extract the information separately.